src/main/java/teammates/ui/automated/PrepareEmailWorkerAction.java

```diff
@@ -0,0 +1,163 @@
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.exception.TeammatesException;
+import teammates.common.util.Assumption;
+import teammates.common.util.EmailType;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.HttpRequestHelper;
+import teammates.common.util.Const.ParamsNames;
+import teammates.logic.core.CommentsLogic;
+import teammates.logic.core.EmailGenerator;
+import teammates.logic.core.EmailSender;
+import teammates.logic.core.FeedbackResponseCommentsLogic;
+import teammates.logic.core.FeedbackSessionsLogic;
+
+/**
+ * Task queue worker action: prepares large number of emails to be sent via task queue.
+ */
+public class PrepareEmailWorkerAction extends AutomatedAction {
+    
+    private final FeedbackSessionsLogic fsLogic = FeedbackSessionsLogic.inst();
+    private final CommentsLogic commentsLogic = CommentsLogic.inst();
+    private final FeedbackResponseCommentsLogic frcLogic = FeedbackResponseCommentsLogic.inst();
+    
+    @Override
+    protected String getActionDescription() {
+        return null;
+    }
+    
+    @Override
+    protected String getActionMessage() {
+        return null;
+    }
+    
+    @Override
+    public void execute() {
+        String emailTypeString = getRequestParamValue(ParamsNames.EMAIL_TYPE);
+        EmailType emailType = EmailType.valueOf(emailTypeString);
+        Assumption.assertNotNull(emailType);
+        
+        String feedbackSessionName = getRequestParamValue(ParamsNames.EMAIL_FEEDBACK);
+        
+        String courseId = getRequestParamValue(ParamsNames.EMAIL_COURSE);
+        Assumption.assertNotNull(courseId);
+        
+        log.info("Prepare email worker activated for: " + HttpRequestHelper.printRequestParameters(request));
+        
+        List<EmailWrapper> emailsToBeSent = prepareEmailsToBeSent(emailType, feedbackSessionName, courseId);
+        
+        if (emailsToBeSent == null) {
+            log.severe("Emails to be sent is null for type: " + emailTypeString);
+            setErrorResponse();
+            return;
+        }
+        
+        try {
+            new EmailSender().sendEmails(emailsToBeSent);
+            doPostProcessingForSuccesfulSend(emailType, feedbackSessionName, courseId);
+        } catch (Exception e) {
+            try {
+                doPostProcessingForUnsuccesfulSend(emailType, courseId);
+            } catch (EntityDoesNotExistException ednee) {
+                log.severe("Unexpected error: " + TeammatesException.toStringWithStackTrace(ednee));
+            }
+            log.severe("Unexpected error: " + TeammatesException.toStringWithStackTrace(e));
+            setErrorResponse();
+        }
+    }
+    
+    private void doPostProcessingForSuccesfulSend(EmailType emailType, String feedbackSessionName, String courseId)
```

Comment:
I think the nested switch-case looks very strange. It might read better if we handle the `PENDING_COMMENT_CLEARED` case as a special case at the top of method:
```
if (Arrays.asList(EmailType.PENDING_COMMENT_CLEARED).contains(emailType)) {
            frcLogic.updateFeedbackResponseCommentsSendingState(
                    courseId, CommentSendingState.SENDING, CommentSendingState.SENT);
            commentsLogic.updateCommentsSendingState(courseId, CommentSendingState.SENDING, CommentSendingState.SENT);
}

FeedbackSessionAttributes feedbackSession = fsLogic.getFeedbackSession(feedbackSessionName, courseId);
        switch (emailType) {
         ....
```